File: SIH/cyber-risk-platform/frontend/src/types/financialRisk.ts
export interface FinancialAssumption {
  id: string;
  organization_id: string;
  category: string;
  name: string;
  value: number;
  unit: string | null;
  currency: string;
  source: string;
  confidence: number;
  effective_from: string;
  effective_until: string | null;
  created_at: string;
  updated_at: string;
}

export interface FinancialBreakdown {
  direct_loss: number;
  data_loss: number;
  business_interruption_loss: number;
  recovery_loss: number;
  customer_impact: number;
  third_party_impact: number;
  regulatory_legal_exposure: number;
  fraud_loss: number;
  reputation_revenue_impact: number;
}

export interface FinancialRiskAssessment {
  id: string;
  organization_id: string;
  asset_id: string;
  risk_score_id: string;
  
  potential_loss: number;
  expected_loss: number;
  annualized_expected_loss: number;
  
  direct_loss: number;
  data_loss: number;
  business_interruption_loss: number;
  recovery_loss: number;
  customer_impact: number;
  third_party_impact: number;
  regulatory_legal_exposure: number;
  fraud_loss: number;
  reputation_revenue_impact: number;
  
  confidence: number;
  data_completeness: number;
  currency: string;
  calculation_version: string;
  
  assumptions_snapshot: Record<string, any> | null;
  metadata: Record<string, any> | null;
  
  calculated_at: string;
  created_at: string;
}

export interface OrganizationFinancialRiskSummary {
  organization_id: string;
  currency: string;
  total_potential_loss: number;
  total_expected_annual_loss: number;
  top_financial_risk_assets: FinancialRiskAssessment[];
  aggregate_breakdown: FinancialBreakdown;
  average_confidence: number;
}

File: SIH/cyber-risk-platform/backend/app/schemas/financial_risk.py
from uuid import UUID
from datetime import datetime
from typing import Optional, Dict, Any, List
from pydantic import BaseModel, Field


# ---------------------------------------------------------------------------
# Financial Assumption Schemas
# ---------------------------------------------------------------------------
class FinancialAssumptionBase(BaseModel):
    category: str = Field(..., max_length=100)
    name: str = Field(..., max_length=255)
    value: float
    unit: Optional[str] = Field(None, max_length=50)
    currency: Optional[str] = Field("INR", max_length=10)
    source: str = Field(..., max_length=255)
    confidence: float = Field(..., ge=0.0, le=100.0)

class FinancialAssumptionCreate(FinancialAssumptionBase):
    organization_id: UUID

class FinancialAssumptionUpdate(BaseModel):
    value: Optional[float] = None
    confidence: Optional[float] = Field(None, ge=0.0, le=100.0)
    source: Optional[str] = Field(None, max_length=255)

class FinancialAssumptionRead(FinancialAssumptionBase):
    id: UUID
    organization_id: UUID
    effective_from: datetime
    effective_until: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


# ---------------------------------------------------------------------------
# Financial Risk Breakdown
# ---------------------------------------------------------------------------
class FinancialBreakdown(BaseModel):
    direct_loss: float
    data_loss: float
    business_interruption_loss: float
    recovery_loss: float
    customer_impact: float
    third_party_impact: float
    regulatory_legal_exposure: float
    fraud_loss: float
    reputation_revenue_impact: float


# ---------------------------------------------------------------------------
# Financial Risk Assessment Schemas
# ---------------------------------------------------------------------------
class FinancialRiskAssessmentBase(BaseModel):
    potential_loss: float = Field(..., ge=0.0)
    expected_loss: float = Field(..., ge=0.0)
    annualized_expected_loss: float = Field(..., ge=0.0)
    
    direct_loss: float = 0.0
    data_loss: float = 0.0
    business_interruption_loss: float = 0.0
    recovery_loss: float = 0.0
    customer_impact: float = 0.0
    third_party_impact: float = 0.0
    regulatory_legal_exposure: float = 0.0
    fraud_loss: float = 0.0
    reputation_revenue_impact: float = 0.0
    
    confidence: float = Field(..., ge=0.0, le=100.0)
    data_completeness: float = Field(..., ge=0.0, le=100.0)
    
    currency: str = "INR"
    calculation_version: str
    
    assumptions_snapshot: Optional[Dict[str, Any]] = None
    financial_metadata: Optional[Dict[str, Any]] = None

class FinancialRiskAssessmentRead(FinancialRiskAssessmentBase):
    id: UUID
    organization_id: UUID
    asset_id: UUID
    risk_score_id: UUID
    calculated_at: datetime
    created_at: datetime

    model_config = {"from_attributes": True, "populate_by_name": True}

class OrganizationFinancialRiskSummary(BaseModel):
    organization_id: UUID
    currency: str = "INR"
    total_potential_loss: float
    total_expected_annual_loss: float
    top_financial_risk_assets: List[FinancialRiskAssessmentRead]
    aggregate_breakdown: FinancialBreakdown
    average_confidence: float
